src/Components/test/testassets/ComponentsApp.Server/Pages/RegularRazor.cshtml

```diff
@@ -0,0 +1,37 @@
+@page
```

Comment:
I wanted to test two things - one we could get to it, and the other to make sure not having a router component does not get us stuck in a place where intercept a link but don't do anything with it. I added a comment stating why we reference the component there.

Reply:
Understood - thanks. If there's any ways of shrinking this file that would be great (e.g., removing all the CSS, the SignalR logging config, the footer, etc) that would be ideal. I'm only concerned that the duplication will make it another thing to remember to update when stuff changes.

Reply:
> make sure not having a router component does not get us stuck in a place where intercept a link but don't do anything with it.

You could also do that easily from the existing E2E entrypoint, since it doesn't have a router except in the "router" tests. If you added a new test component like any of the non-router tests, that would represent the scenario where there can be `<a>` tags and no `<Router>`.

Then the other case is just "we could get to it", right? That could be a trivial static HTML file.

Reply:
Maybe I'm missing something, but the App in the server side test always has a router?

Reply:
It doesn't always have one. The only `<Router>` in `BasicTestApp` is in `RouterTest\TestRouter.razor`, which is not used except if you specifically pick "Router" from the test selector. Note how the only components with `@page` directives are the ones under `RouterTest\*`.

Reply:
Oh wait, you're talking about `ComponentsApp.Server`. We're trying not to put E2E tests there (in fact we have [a bug tracking the work to take existing ones out](https://github.com/aspnet/AspNetCore/issues/8992)).

Is it possible to put all the E2E tests for this in `BasicTestApp`? This one works as I described, with the router being optional. Anything in that app can run on both the server and the client. AFAIK the "defer link interception" feature should work identically in server-side and client-side Blazor, so the same test case should apply to either.

Reply:
> Is it possible to put all the E2E tests for this in BasicTestApp?

I guess we already have a scenario of ensuring non-Blazor links are followed in BasicTestApp. I wanted to add a test that captures the user reported scenario. The external link being Razor Pages isn't super relevant here I suppose.